server/accounts.go

```diff
@@ -4114,18 +4116,73 @@ type transform struct {
 	src, dest string
 	dtoks     []string
 	stoks     []string
-	dtpi      []int8
+	dtpi      [][]int // destination token position indexes
+	dtpinp    []int32 // destination token position index number of partitions
 }
 
-// Helper to pull raw place holder index. Returns -1 if not a place holder.
-func placeHolderIndex(token string) int {
-	if len(token) > 1 && token[0] == '$' {
-		var tp int
-		if n, err := fmt.Sscanf(token, "$%d", &tp); err == nil && n == 1 {
-			return tp
+func getMappingFunctionArgs(functionName string, token string) ([]string, error) {
+	regEx := `{{` + functionName + ` *\((.*)\)}}`
```

Comment:
I would create 2 regex'es (as var at top of this file) and pass those regex as the func argument (instead of name). The compile should be done at a minimum. I checked the calling stack of this and it is not only called during processing of config, it can happen when processing a sub, etc.. so I would try to reduce the cost here.